angel_system/activity_hmm/core.py

```diff
@@ -275,37 +279,63 @@ def __init__(self, dt, class_str, med_class_duration,
 
         # Indices into N_-length arrays (e.g., class_str_) that correspond to the
         # original N-length arrays (e.g., class_str).
-        not_bckg_mask_ = ~bckg_mask.copy()
+        not_bckg_mask = ~bckg_mask
 
+        # Mask capturing the elements of X_ that correspond to the elements of
+        # X.
+        orig_mask = not_bckg_mask.copy()
+        orig_mask[0] = True
 
-        if class_mean_conf is not None and class_std_conf is not None:
+        if class_mean_conf is not None:
             class_mean_conf = np.array(class_mean_conf)
-            class_std_conf = np.array(class_std_conf)
 
-            # We define an N x N mean matrix where element (i, j) is the mean value
-            # emmitted for class j when class i is the state.
+            # We define an N x N mean matrix where element (i, j) is the mean
+            # value emmitted for class j when class i is the state.
             if np.any(class_mean_conf > 1):
                 raise Exception('\'class_mean_conf\' must be between 0-1')
 
-            class_conf_mean_mat = np.zeros((N_, N_))
-            sphere_conf_cov_mat = np.zeros(N_)
-
-            # Square to turn from std to cov. We'll reuse the name.
-            class_std_conf = class_std_conf**2
+            if np.ndim(class_mean_conf) == 1:
+                class_mean_conf = np.diag(class_mean_conf)
+            elif np.ndim(class_mean_conf) > 2:
+                raise Exception('np.ndim(class_mean_conf) must be 1 or 2')
```

Comment:
Generally do not use base-class exceptions, [concrete exceptions](https://docs.python.org/3/library/exceptions.html#concrete-exceptions) should be used if looking for a builtin, define your own exception, or use an assert.

Reply:
Addressed.